src/reporter/index.ts

```diff
@@ -635,4 +648,18 @@ export default class Reporter {
 
         (this.taskInfo.pendingTaskDonePromise.resolve as Function)();
     }
+
+    private async _onReportDataHandler ({ testRun, data }: ReportDataEventArgs): Promise<void> {
+        if (!this.taskInfo)
+            return;
+
+        await this.dispatchToPlugin({
```

Comment:
Reporting should be optional. If target reporter doesn't implement the `reportData` method, dispatching should not be called.